File: src/ipc/transfer.ts
import { safeInvoke } from './client';

export async function cancelTransfer(transferId: string): Promise<void> {
  await safeInvoke('transfer_cancel', { transferId });
}

export async function cancelAllTransfers(connectionId: string): Promise<void> {
  await safeInvoke('transfer_cancel_all', { connectionId });
}

File: src-tauri/src/commands/transfer.rs
use std::sync::Arc;

use tauri::State;

use crate::services::transfer_cancel::TransferCancelRegistry;

#[tauri::command]
pub fn transfer_cancel(
    registry: State<'_, Arc<TransferCancelRegistry>>,
    transfer_id: String,
) {
    registry.cancel(&transfer_id);
}

#[tauri::command]
pub fn transfer_cancel_all(
    registry: State<'_, Arc<TransferCancelRegistry>>,
    connection_id: String,
) {
    registry.cancel_all(&connection_id);
}
